### ISSUE TITLE
Cloud Build IOS - linker error

### ISSUE THREAD

**Issue by oosmoxiecode** [firebase/quickstart-unity#112]
I´m having trouble using Unity Cloud Build to build an ios version of my project. This worked fine until a few weeks ago.

The log now says:
> 17507:        [xcode] ld: library not found for -lz}
> 17508:        [xcode] clang: error: linker command failed with exit code 1 (use -v to see invocation)

Notice the trailing "}".
This seems related to firebase and the IOSResolver as far as I can tell. 

This thread might give more insight:
https://groups.google.com/forum/?hl=ja#!topic/firebase-talk/jxur4oWjE2U

Something that can be solved?
Thanks!

**Comment by stewartmiles** [comment]
Yep we're working on this.  The problem is in the parser here https://github.com/googlesamples/unity-jar-resolver/blob/6e56d2d22fb246e9ee8ace390dd04a379fb8a114/source/IOSResolver/src/IOSResolver.cs#L317 which is tripped up by FirebaseInstanceID

Since this is caused by a transitive dependency upon FirebaseInstanceID you can add the pod with the modulemap that does *not* break our poor excuse for a modulemap parser.

Open up `Firebase/Editor/AppDeps.cs` and add the following line to the iOS section to force the use of the FirebaseInstanceID pod at version 2.0.1.

```
        Google.VersionHandler.InvokeStaticMethod(
            iosResolver, "AddPod",
            new object[] { "FirebaseInstanceID" }, 
            new Dictionary<string, object>() { 
                { "version", "2.0.1" },
                { "minTargetSdk", null },
                { "sources", null }
            });
```

Cheers,
Stewart

**Comment by stewartmiles** [comment]
Issue filed against the iOS resolver here https://github.com/googlesamples/unity-jar-resolver/issues/65

**Comment by oosmoxiecode** [comment]
Thanks Stewart!

I tested to add those lines to `AppDeps.cs`, then Cloud Build failed differently.
Like this:
> 8456: [Unity] Player export failed. Reason: iOS framework addition failed due to a Cocoapods installation failure. This will will likely result in an non-functional Xcode project.
> 8457: [Unity] After the failure, "pod repo update" was executed and succeeded. "pod install" was then attempted again, and still failed. This may be due to a broken Cocoapods installation. See: https://guides.cocoapods.org/using/troubleshooting.html for potential solutions.
> 8458: [Unity] pod install output:
> 8459: [Unity] Setting up CocoaPods master repo
> 8460: [Unity]   $ /usr/bin/git clone https://github.com/CocoaPods/Specs.git master --progress
> 8461: [Unity] Setup completed
> 8462: [Unity] Analyzing dependencies
> 8463: [Unity] [!] Unable to satisfy the following requirements:
> 8464: [Unity] - `Firebase/Core (= 4.1.0)` required by `Podfile`
> 8465: [Unity] - `Firebase/Core (= 4.0.4)` required by `Firebase/Analytics (4.0.4)`
> 8466: [Unity] Specs satisfying the `Firebase/Core (= 4.1.0), Firebase/Core (= 4.0.4)` dependency were found, but they required a higher minimum deployment target.

Any ideas how to solve that?
Thanks!

**Comment by stewartmiles** [comment]
It looks like your project is specifying both Firebase/Core 4.1.0 and Firebase/Analytics 4.0.4.  If you're just using Firebase's plugins I think you should check your `Editor/*Deps.cs` files for references to `Firebase/*` pods at version 4.0.4 because they sound like stale files in the plugin.

**Comment by TwitRco** [comment]
**koek-koek**

**Comment by oosmoxiecode** [comment]
I tested to add the same thing to `AnalyticsDeps.cs` and got it to build.
Thanks again!

**Comment by stewartmiles** [comment]
https://github.com/googlesamples/unity-jar-resolver version 1.2.50 has a fix for the issue, we'll roll it up into the Firebase SDK shortly.

**Comment by stewartmiles** [comment]
The Play Services Resolver fix is now available with other improvements in the 4.2.0 release of the Firebase Unity SDK.  https://firebase.google.com/support/release-notes/unity#4.2.0
